### DISCUSSION TITLE
Saving data as .dm

### DISCUSSION THREAD

**Discussion by HanHsuanWu** [nion-software/nionswift#911] [Q&A] [answered]
Hi guys,

Does nion swift no longer support saving files in dm format?
It seems like the ImportExportManager does not include dm extension.
https://github.com/nion-software/nionswift/blob/81d02a6271752cf5f7acfb0965c6a3e7b2abe829/nion/swift/model/ImportExportManager.py#L775-L788

**Comment by cmeyer** [comment]
The .dm3 and .dm4 import/export gets loaded with the `nionswift-io` package, which must be installed in your Python environment along with `nionswift`.

- https://github.com/nion-software/nionswift-io
- https://github.com/nion-software/nionswift-io/tree/master/nionswift_plugin/DM_IO

One issue is on our current to-do list:
- https://github.com/nion-software/nionswift-io/issues/16

Do you have another issue?

**Comment by HanHsuanWu** [comment]
Hi Chris,

Thanks for you previous reply. I am trying to follow the Interactive Scripting Guide. The example given below from the guide seems to cause problem while printing.
![image](https://user-images.githubusercontent.com/112028355/223594957-5954a7b2-43cf-42a3-9b10-ce5ad1ff4f72.png)


The example interactive script from https://github.com/nion-software/nionswift/blob/e874ae2a0828f32d637a9c00142910036a35325a/docs/api/interactive.rst:

from nion.typeshed import Interactive_1_0 as Interactive
from nion.typeshed import API_1_0 as API
from nion.typeshed import UI_1_0 as UI

def do_something(interactive: Interactive, api: API):
    library = api.library
    target_data_item = api.application.document_windows[0].target_data_item
    print('Sum of target image: ', numpy.sum(target_data_item.data))

def script_main(api_broker):
    interactive = api_broker.get_interactive(Interactive.version)  # type: Interactive
    api = api_broker.get_api(API.version, UI.version)  # type: API
    do_something(interactive, api)

Update:
https://github.com/nion-software/nionswift/blob/6262f2342432966cd688495fec6bf7a437da019d/nion/swift/ScriptsDialog.py#L775-L778
It seems like RunScriptDialog.print() doesn't work when comma is used in print("message1","message2"), but multiple print("message1") print("message2") is fine.

**Reply by tylerharvey** [reply, marked as answer]
You can get around the single-argument limit of RunScriptDialog.print() by using [f-strings.](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings).

**Reply by HanHsuanWu** [reply]
Thanks Tyler lol!
